src/services/infra/InfraService.ts

```diff
@@ -229,6 +260,159 @@ export default class InfraService {
     return ok(dnsRecord)
   }
 
+  launchSiteFromCms({
+    siteName,
+    primaryDomain,
+    useWww,
+    email,
+  }: LaunchSiteFromCMSParams): ResultAsync<
+    SiteLaunchCreateParams,
+    AmplifyError | SiteLaunchError | MissingUserEmailError
+  > {
+    // prepare site launch params
+    return ResultAsync.fromPromise(
```

Comment:
not a blocker, but i feel like if the precondition to this function is to have a valid user email, we might want to strengthen the params from `email: string` -> `user: User`

Reply:
i think my concern here is that this feels like a router responsibility (to validate parameters for services) rather than a service responsibility. 

i envision routers doing validation etc prior to passing on to the service, so that services can be strict about their pre-conditions. hence, i'd prefer this to be separated! (only saw 1 call site)

we should also wrap the methods for `userService` to return a `Result` but out of scope

Reply:
> i think my concern here is that this feels like a router responsibility (to validate parameters for services) rather than a service responsibility

just curious here, shoudnt our mental model be router should verify params rather than validate since this validation of parameters for services here requires a call to our repositories?